src/slowmokit/methods/linear_model/logistic_regression/logistic_regression.cpp

```diff
@@ -0,0 +1,200 @@
+    {
+      std::cout << "Epoch: " << i << " "
+                << "Loss: " << l << std::endl;
+    }
+
+    std::vector<int> prm(n); // Making vector of size 'n' so that we can
+                             // randomly shuffle number from 0-(n-1) and choose
+                             // values in random order to perform regression
+    for (int j = 0; j < n; j++)
+    {
+      prm[j] = j;
+    }
+
+    std::random_shuffle(prm.begin(),
+                        prm.end()); // random shuffling of prm vector
+
+    while (!prm.empty()) // till prm not becomes empty keep iterating over the
+                         // while loop
+    {
+      int frnt = prm.back(); // gives last value of vector "prm"
+      std::vector<double> xiHat =
+          x[frnt]; // now we take random value of x according to the frnt value
+                   // obtained!
+      std::vector<int> yi = y[frnt]; //   Gives y value corresponding to xiHat
+      std::vector<double> zi(
+          numClasses, 0.0); // Initialization of zi vector which contains the
+                            // predicted value of xiHat(one hot encoded)
+                            //   Formula for z[i] is written below
+
+      // z[i] = weight[ouput class][j-th weight] * x[i-th row][j-th feature]
+
+      for (int j = 0; j < numClasses; j++)
+      { // Iterating over classes of output
+        for (int k = 0; k < xiHat.size(); k++)
+        { // Iterating over x[i]
+          zi[j] += beta[j][k] *
+                   xiHat[k]; // Here we are performing  |  z[i] = weight[ouput
+                             // class][j-th weight] * x[i-th row][j-th feature]
+        }
+      }
+      zi = softmax(zi); // Now we are computing here the softmax of all one hot
+                        // encoded values obtained for particular xiHat
+
+      std::vector<std::vector<double>> gradLi( // Initializing gradient descent
+          numClasses, std::vector<double>(xiHat.size()));
+      for (int j = 0; j < numClasses; j++) // iterating over all classes
+      {
+        for (int k = 0; k < xiHat.size();
+             k++) // iterating over all xiHat values
+        {
+          gradLi[j][k] = (zi[j] - yi[j]) *
+                         xiHat[k]; // here we are computing gradient descent
+          beta[j][k] -= alpha * gradLi[j][k]; // Changing value of beta
+                                              // according to the computed loss
+        }
+      }
+
+      prm.pop_back(); // deleting prm last value to take next random value for
+                      // xiHat
+    }
+  }
+  return beta; // finally return weights computed after all the epochs
+};
+
+template<class T>
+// training function
+// x -> training 2-d vector double values
+// y - > training 1-d int values
+// alpha -> double which is going to be multiply with gradient descent
+// numEpochs -> number of epochs
+// verbose -> whether to show which epoch is going on
+void LogisticRegression<T>::train(std::vector<std::vector<T>> x,
```

Comment:
We'll be assuming that user won't provide onehot-encoded labels. You need to do that inside your code

Reply:
You need not write your own one-hot encoder. Use the one already written in our library. Also change the name of `train` function to `fit`